### DISCUSSION TITLE
How do you change the default dashboard for lazy.vim?

### DISCUSSION THREAD

**Discussion by voodookiidoo** [LazyVim/LazyVim#4717] [Q&A]
I'm not really good at lua, but looking at examples, i thought i understood something.
`{
    "nvimdev/dashboard-nvim",
    config = function()
      require("plugins.dashboard")
    end,
    dependencies = { { "nvim-tree/nvim-web-devicons" } },
  },`
and inside dashboard.lua file i have a simple config like this
`return {
  opts = {
    theme = "doom",
  },
}`
but nothing changes, despite me playing with the settings as much as i want. it still welcomes me with the default lazyvim screen. Please help :(

**Comment by Jacobus-afk** [comment]
Hi @voodookiidoo 

Have you  looked at this discussion?

https://github.com/LazyVim/LazyVim/discussions/742#discussioncomment-8257940

**Reply by voodookiidoo** [reply]
yep, even tried to do it with alpa-nvim, i don't understand what might be the problem

**Comment by Jacobus-afk** [comment]
as mentioned in that discussion, it's important to place extensions to plugins in the `~/.config/nvim/lua/plugins/` directory

i created a file inside there called **extend-dashboard.lua** (the name doesn't really matter, as long as its in the plugins directory)

contents of extend-dashboard.lua:

```
return {
	{
		"nvimdev/dashboard-nvim",
		event = "VimEnter",
		opts = function(_, opts)
			local logo = [[
                                                                                                                     
        ,--,                                                                                                         
      ,--.'|              ,--,      ,--,                            .---.                        ,--,                
   ,--,  | :            ,--.'|    ,--.'|                           /. ./|                      ,--.'|          ,---, 
,---.'|  : '            |  | :    |  | :       ,---.           .--'.  ' ;    ,---.     __  ,-. |  | :        ,---.'| 
|   | : _' |            :  : '    :  : '      '   ,'\         /__./ \ : |   '   ,'\  ,' ,'/ /| :  : '        |   | : 
:   : |.'  |    ,---.   |  ' |    |  ' |     /   /   |    .--'.  '   \' .  /   /   | '  | |' | |  ' |        |   | | 
|   ' '  ; :   /     \  '  | |    '  | |    .   ; ,. :   /___/ \ |    ' ' .   ; ,. : |  |   ,' '  | |      ,--.__| | 
'   |  .'. |  /    /  | |  | :    |  | :    '   | |: :   ;   \  \;      : '   | |: : '  :  /   |  | :     /   ,'   | 
|   | :  | ' .    ' / | '  : |__  '  : |__  '   | .; :    \   ;  `      | '   | .; : |  | '    '  : |__  .   '  /  | 
'   : |  : ; '   ;   /| |  | '.'| |  | '.'| |   :    |     .   \    .\  ; |   :    | ;  : |    |  | '.'| '   ; |:  | 
|   | '  ,/  '   |  / | ;  :    ; ;  :    ;  \   \  /       \   \   ' \ |  \   \  /  |  , ;    ;  :    ; |   | '/  ' 
;   : ;--'   |   :    | |  ,   /  |  ,   /    `----'         :   '  |--"    `----'    ---'     |  ,   /  |   :    :| 
|   ,/        \   \  /   ---`-'    ---`-'                     \   \ ;                           ---`-'    \   \  /   
'---'          `----'                                          '---"                                       `----'    
                                                                                                                     
            ]]

			logo = string.rep("\n", 8) .. logo .. "\n\n"
			opts.config.header = vim.split(logo, "\n")
			opts.theme = "hyper"
		end,
	},
}
```

this is how my dashboard looks then
![image](https://github.com/user-attachments/assets/ec2f112c-2f01-4d96-9233-aadad05a77ce)

**Comment by cerebrl** [comment]
If you would like a working example: https://github.com/cerebrl/nvim/blob/main/lua/plugins/dashboard.lua

**Comment by dpetka2001** [comment]
From your initial post, I'm assuming you have 2 different files for `dashboard`. The first file has 
```lua
return {
  {
    "nvimdev/dashboard-nvim",
    dependencies = {
      { "nvim-tree/nvim-web-devicons" },
    },
    config = function()
      require("plugins.dashboard")
    end,
  },
}
```
And the second file is also in `/lua/plugins/` directory with contents
```lua
return {
  opts = {
    theme = "doom",
  },
}
```
But this is not a valid plugin spec. You should either follow the example above as the other user said by putting everything in one file under `/lua/plugins/` directory or if you still want to have 2 files for your own reasons, you could do the following:
1. Put the first file in `/lua/plugins/` directory (for example `/lua/plugins/dashboard.lua`) with contents
```lua
return {
  {
    "nvimdev/dashboard-nvim",
    dependencies = {
      { "nvim-tree/nvim-web-devicons" },
    },
    config = function()
      require("your_username.dashboard")
    end,
  },
}
```
Notice that I changed the require from `plugins.dashboard` to `your_username.dashboard`. 
2. You have to create a directory `/lua/your_username/` and put there your other dashboard file (for example `/lua/your_username/dashboard.lua`) or whatever other unique namespace of your choosing (instead of `your_username` it can be `whatever_else_you_want`, but you should use that namespace in the require of the first file) with contents
```lua
local opts = { 
  theme = "doom", 
}
require("dashboard").setup(opts)
```
I believe that should work if you want to keep having your configs in separate files. One file for calling the plugin spec (`/lua/plugins/dashboard.lua`) and another file where your options for the plugin will exist (`/lua/your_username/dashboard.lua`)

**Reply by voodookiidoo** [reply]
> From your initial post, I'm assuming you have 2 different files for `dashboard`. The first file has
> 
> ```lua
> return {
>   {
>     "nvimdev/dashboard-nvim",
>     dependencies = {
>       { "nvim-tree/nvim-web-devicons" },
>     },
>     config = function()
>       require("plugins.dashboard")
>     end,
>   },
> }
> ```
> 
> And the second file is also in `/lua/plugins/` directory with contents
> 
> ```lua
> return {
>   opts = {
>     theme = "doom",
>   },
> }
> ```
> 
> But this is not a valid plugin spec. You should either follow the example above as the other user said by putting everything in one file under `/lua/plugins/` directory or if you still want to have 2 files for your own reasons, you could do the following:
> 
> 1. Put the first file in `/lua/plugins/` directory (for example `/lua/plugins/dashboard.lua`) with contents
> 
> ```lua
> return {
>   {
>     "nvimdev/dashboard-nvim",
>     dependencies = {
>       { "nvim-tree/nvim-web-devicons" },
>     },
>     config = function()
>       require("your_username.dashboard")
>     end,
>   },
> }
> ```
> 
> Notice that I changed the require from `plugins.dashboard` to `your_username.dashboard`. 2. You have to create a directory `/lua/your_username/` and put there your other dashboard file (for example `/lua/your_username/dashboard.lua`) or whatever other unique namespace of your choosing (instead of `your_username` it can be `whatever_else_you_want`, but you should use that namespace in the require of the first file) with contents
> 
> ```lua
> local opts = { 
>   theme = "doom", 
> }
> require("dashboard").setup(opts)
> ```
> 
> I believe that should work if you want to keep having your configs in separate files. One file for calling the plugin spec (`/lua/plugins/dashboard.lua`) and another file where your options for the plugin will exist (`/lua/your_username/dashboard.lua`)


This is my config dir structure
<img width="420" alt="image" src="https://github.com/user-attachments/assets/349218f5-9db3-4d3c-b70f-7efbf924dbc7">

I'm trying to specify the dasbhoard configuration right inside the lazy.lya - but it doesn't seem to have any effect :(

<img width="806" alt="image" src="https://github.com/user-attachments/assets/3643ae71-70a8-4226-aaa2-2bd9f47170d6">

I have no idea what to do, this is the hardest configuring process i have ever done (maybe cause i dont know lua but still)

Please help

**Reply by folke** [reply]
That is not correct and won't work.
Do as @dpetka2001 says and create a file under plugins.
You can't setup lazy more than once and if that is what in your lazy.lua file, then you have removed loading of LazyVim and everything.

Start with reading the docs of `lazy.nvim` the plugin manager and then the docs for lazyvim.
Both have a separate website with all the info you need to get started.
